backend/tournesol/tests/test_exports.py

```diff
@@ -434,3 +445,55 @@ def test_all_export_sorts_by_username(self):
                 usernames = [row["public_username"] for row in rows]
                 self.assertEqual("a", usernames[0])
                 self.assertEqual("z", usernames[-1])
+
+    def test_all_export_comparisons_sorted_by_username_and_weekdate(self):
+        first_user = UserFactory(username="zoe")
+        last_user = UserFactory(username="alain")
+        
+        # comparisons#1, on a given Monday
+        with MockNow.Context(datetime(2023, 1, 16, tzinfo=timezone.utc)) as mock_date_1:
+            self.add_comparison(user=first_user, is_public=True)
+            self.add_comparison(user=last_user, is_public=True)
+
+        # comparisons#2, the same week on a Sunday
+        with MockNow.Context(datetime(2023, 1, 22, tzinfo=timezone.utc)) as mock_date_2:
+            self.add_comparison(user=first_user, is_public=True)
+            self.add_comparison(user=last_user, is_public=True)
+
+        # comparisons#3. random date in current week
+        random_date_this_week = datetime.now()
+        random_date_this_week = random_date_this_week - timedelta(random_date_this_week.weekday()) + timedelta(random.randint(0, 6))
+        self.add_comparison(user=first_user, is_public=True)
+        self.add_comparison(user=last_user, is_public=True)
+
+        call_command("create_dataset")
+
+        response = self.client.get("/exports/all/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.headers["Content-Type"], "application/zip")
+        root = self.extract_export_root(response)
+
+        zip_content = io.BytesIO(response.content)
+
+        with zipfile.ZipFile(zip_content, "r") as zip_file:
+            with zip_file.open(root + "/comparisons.csv", "r") as file:
```

Comment:
Maybe consider using Pathlib

Reply:
@aidanjungo I simply copied-pasted from another test 😅

Reply:
I'm touching these same tests in another PR. Best to leave it like this for now I think

Reply:
Yes I will do another PR to change that everywhere